packages/js/src/workouts/components/Modal.js

```diff
@@ -0,0 +1,85 @@
+import { Dialog, Transition } from "@headlessui/react";
+import { XIcon } from "@heroicons/react/outline";
+import { Fragment } from "@wordpress/element";
+import { __ } from "@wordpress/i18n";
+import { noop } from "lodash";
+import { PropTypes } from "prop-types";
+
+/* eslint-disable max-len */
+/**
+ * Modal component.
+ *
+ * @param {Object} props The props.
+ * @param {boolean} props.isOpen Whether the modal is open.
+ * @param {function} [props.handleClose] Function that is called when the user wants to close the modal.
+ * @param {boolean} [props.hasCloseButton=true] Whether the modal has a close button.
+ * @param {JSX.node} props.children Contents of the modal.
+ *
+ * @returns {JSX.Element} The modal element.
+ */
+export default function Modal( { isOpen, handleClose, hasCloseButton, children } ) {
```

Comment:
Perhaps we could call the component (and possibly the filename) `TailwindModal` as we already have some other modals within our plugin?